connect/api/src/main/java/org/apache/kafka/connect/rest/extension/ConnectRestExtension.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest.extension;
+
+import org.apache.kafka.common.Configurable;
+
+import java.io.Closeable;
+
+public interface ConnectRestExtension extends Configurable, Closeable {
+
+    void register(ConnectRestExtensionContext restPluginContext);
```

Comment:
Missing JavaDoc on the type and method. IMO, the JavaDoc on the interface should fully describe how to provide an implementation (by implementing this class, but what about other interfaces), how to package it (e.g., Java Service Provider file), and how to install it (put it on the plugin path).

It should also go into detail about how Connect uses this interface, when implementations are instantiated and when the register method is called, and when the close method is called.

What exceptions can be thrown by this method? Will the supplied context ever be null? What are the behaviors that are expected/allowed? What happens when a resource is already registered?